test-data/unit/check-selftype.test

```diff
@@ -354,16 +354,21 @@ class E:
 [case testSelfTypeProperty]
 from typing import TypeVar
 
-T = TypeVar('T', bound='A')
+Q = TypeVar('Q')
+T = TypeVar('T', bound='X')
 
-class A:
+class X:
     @property
-    def member(self: T) -> T:
-        pass
+    def __members__(self: Q) -> Q: return self
+
+class A(X):
+    @property
+    def member(self: T) -> T: return self
 
 class B(A):
     pass
 
+reveal_type(X().__members__)  # E: Revealed type is '__main__.X*'
```

Comment:
Would it make sense to add a test case similar to this, but so that `X` would a subtype of `Tuple`?